src/components/sectionList/listActions/DeleteAction.tsx

```diff
@@ -0,0 +1,90 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    IconDelete16,
+    MenuItem,
+    Modal,
+    ModalActions,
+    ModalTitle,
+} from '@dhis2/ui'
+import React, { useState } from 'react'
+import { TOOLTIPS } from '../../../lib'
+import { TooltipWrapper } from '../../tooltip'
+
+export function DeleteAction({
+    deletable,
+    modelType,
+    onClick,
+    onCancel,
+}: {
+    deletable: boolean
+    modelType: string
+    onClick: () => void
+    onCancel: () => void
+}) {
+    const [showConfirmationDialog, setShowConfirmationDialog] = useState(false)
+
+    return (
+        <>
+            <TooltipWrapper
+                condition={!deletable}
+                content={TOOLTIPS.noDeleteAccess}
+            >
+                <MenuItem
+                    dense
+                    disabled={!deletable}
+                    label={i18n.t('Delete')}
+                    icon={<IconDelete16 />}
+                    onClick={() => setShowConfirmationDialog(true)}
+                />
+            </TooltipWrapper>
+
+            {showConfirmationDialog && (
+                <ConfirmationDialog
```

Comment:
I don't think it's a good idea to have the modal here. All of this is rendered inside a condintally rendered popper and could unmount, right?

Reply:
> All of this is rendered inside a condintally rendered popper and could unmount, right?

When could an unmount happen? Right now, I don't see a scenario where this could happen.

> I don't think it's a good idea to have the modal here.

The reason why I moved the dialog here is to make things more simple. The concept of deleting a single item is now within the `DeleteAction` component only. If another component had to render the modal, it's complect that component as it'd have multiple purposes although it shouldn't care / be aware, except for whether deletion was successful so the list can be refetched (which the `DeleteAction` component now doesn't need to know anything about).